packages/web-app-files/src/helpers/breadcrumbs.ts

```diff
@@ -1,31 +1,20 @@
-import { bus } from 'web-pkg/src/instance'
-
 interface BreadcrumbItem {
   text: string
   to?: string
+
   onClick?()
 }
 
-export const breadcrumbsFromPath = (routePath: string, resourcePath: string = ''): BreadcrumbItem[] => {
-
-  const basePaths =
-    '/' +
-    decodeURIComponent(routePath)
-      .replace(resourcePath, '')
-      .split('/')
-      .filter(Boolean)
-      .join('/')
+export const breadcrumbsFromPath = (
+  currentPath: string,
+  resourcePath: string
+): BreadcrumbItem[] => {
+  const pathSplit = (p = '') => p.split('/').filter(Boolean)
+  const current = pathSplit(currentPath)
+  const resource = pathSplit(resourcePath)
 
-  return [basePaths, ...resourcePath.split('/').filter(Boolean)].reduce(
-    (acc, rawItem, i, rawItems) => {
-      const to = [(acc[i - 1] || {}).to, rawItem].filter(Boolean).join('/')
-      acc.push({
-        text: rawItem,
-        // onClick: () => bus.publish('app.files.list.load', to.replace(basePaths, '') || '/'),
-        to
-      })
-      return acc
-    },
-    []
-  )
+  return resource.map((text, i) => ({
+    text,
+    to: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/')
```

Comment:
We'll have to resort to `onClick: () => bus.publish('app.files.list.load')` for the last item (otherwise the breadcrumbs don't render properly 🙄 )

Reply:
Fixed with help from @kulmann